graphql_compiler/compiler/helpers.py

```diff
@@ -196,16 +198,12 @@ def revisit(self):
 
     def get_location_name(self):
         """Return a tuple of a unique name of the Location, and the current field name (or None)."""
-        mark_name = u'__'.join(self.query_path) + u'___' + unicode(self.visit_counter)
+        mark_name = u'__'.join(self.query_path) + u'___' + six.text_type(self.visit_counter)
         return (mark_name, self.field)
 
-    def __unicode__(self):
-        """Return a human-readable unicode representation of the Location object."""
-        return u'Location({}, {}, {})'.format(self.query_path, self.field, self.visit_counter)
-
     def __str__(self):
         """Return a human-readable str representation of the Location object."""
-        return self.__unicode__().encode('utf-8')
+        return u'Location({}, {}, {})'.format(self.query_path, self.field, self.visit_counter)
```

Comment:
Isn't this technically incorrect for py2? In py2, `__str__` should return bytes and not unicode, I think.

Reply:
Or does the magic `python_2_unicode_compatible` solve this?

Reply:
Yah, magic solution.  I did not know this was a problem until I saw you take care of it, and was surprised to find there's a nice solution:

> A class decorator that takes a class defining a __str__ method. On Python 3, the decorator does nothing. On Python 2, it aliases the __str__ method to __unicode__ and creates a new __str__ method that returns the result of __unicode__() encoded with UTF-8.

https://pythonhosted.org/six/#six.python_2_unicode_compatible